connectors/sources/sharepoint_online.py

```diff
@@ -1286,29 +1393,36 @@ async def get_docs(self, filtering=None):
                 site_collection["siteCollection"]["hostname"],
                 self.configuration["site_collections"],
             ):
-                access_control = await self._site_access_control(site)
-                yield self._decorate_with_access_control(site, access_control), None
+                site_access_control = await self._site_access_control(site)
+                yield self._decorate_with_access_control(
+                    site, site_access_control
+                ), None
 
                 async for site_drive in self.site_drives(site):
                     yield self._decorate_with_access_control(
-                        site_drive, access_control
+                        site_drive, site_access_control
                     ), None
 
                     async for page in self.client.drive_items(site_drive["id"]):
-                        for drive_item in page:
-                            drive_item["_id"] = drive_item["id"]
-                            drive_item["object_type"] = "drive_item"
-                            drive_item["_timestamp"] = drive_item[
-                                "lastModifiedDateTime"
-                            ]
-
-                            drive_item = self._decorate_with_access_control(
-                                drive_item, access_control
-                            )
-
-                            yield drive_item, self.download_function(
-                                drive_item, max_drive_item_age
-                            )
+                        for drive_items_batch in iterable_batches_generator(
+                            page.items, SPO_API_MAX_BATCH_SIZE
```

Comment:
Given `SPO_API_MAX_BATCH_SIZE = 20`, to gain the maximum performance, we'd better get `page` from `drive_items` at a multiple of 20.

The delta query for drive item supports `$top` param, but seems we didn't specify it at all. I did a search in the official doc, but I couldn't find the default and maximum value for this param

Reply:
Good point, I would also keep this for now separate and create a corresponding issue to deal with it later as getting the configurable permissions out is prio #1 right now.